src/main/java/io/supertokens/session/Session.java

```diff
@@ -336,21 +363,29 @@ private static SessionInformationHolder refreshSessionHelper(Main main, String r
 
                         if (sessionInfo.refreshTokenHash2.equals(Utils.hashSHA256(Utils.hashSHA256(refreshToken)))) {
                             // at this point, the input refresh token is the parent one.
+
+                            JsonObject grantPayload = sessionInfo.grants;
+                            // TODO(mihaly): we could skip updating this in DB until a value is set
+                            if (grantPayload == null) {
+                                grantPayload = new JsonObject();
+                                storage.updateSessionGrantPayload_Transaction(con, sessionHandle,
+                                        grantPayload.toString());
+                            }
```

Comment:
I don't understand this. To discuss

Reply:
The comment just says that we could skip updating the DB until we have a value to set instead of a default empty object, but I don't think that'd be a good idea. I'll remove the comment.